sdk/python/kfmd/metadata.py

```diff
@@ -82,21 +86,32 @@ def log(self, artifact):
     This method expects `artifact` to have
       - ARTIFACT_TYPE_NAME stirng field the form of
         /artifact_types/<namespace>/<name>.
-      - serialization() method to return a swagger_client.ApiArtifact.
+      - serialization() method to return a openapi_client.MlMetadataArtifact.
 
-    This method will set artifact._id and artifact._create_time.
+    This method will set artifact.id.
     """
 
     # TODO(zhenghui): log this artifact as the input or output of an execution.
     serialization = artifact.serialization()
-    if serialization.workspace.name == None:
-      serialization.workspace.name = self.workspace.name
+    if serialization.custom_properties == None:
+          serialization.custom_properties = {}
+    if self.WORKSPACE_PROPERTY_NAME in serialization.custom_properties:
+          raise ValueError("custom_properties contains reserved key %s"
+                           % self.WORKSPACE_PROPERTY_NAME)
+    if self.RUN_PROPERTY_NAME in serialization.custom_properties:
+      raise ValueError("custom_properties contains reserved key %s"
+                       % self.RUN_PROPERTY_NAME)
+    serialization.custom_properties[
```

Comment:
Would you mind increasing the linebreak length? This is a little hard to read.

Reply:
Actually, I prefer it this way. I recommend we follow the style guide prescription here:
http://google.github.io/styleguide/pyguide.html#32-line-length